### PR TITLE
py-netcdf: update

### PR DESCRIPTION
**Pull request by citibeth**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-netcdf/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class PyNetcdf(PythonPackage):
+    """Python interface to the netCDF Library."""
+    homepage = "http://unidata.github.io/netcdf4-python"
+    url      = "https://github.com/Unidata/netcdf4-python/tarball/v1.2.3.1rel"
+
+    version('1.2.3.1', '4fc4320d4f2a77b894ebf8da1c9895af')
+
+    depends_on('py-numpy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netcdf/package.py]: `py-numpy@1.7:`

File: var/spack/repos/builtin/packages/py-netcdf/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-cython', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netcdf/package.py]: `py-cython@0.19:`

File: var/spack/repos/builtin/packages/py-netcdf/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('netcdf')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netcdf/package.py]: I'm seeing a lot of other dependencies here:
```python
HDF5_dir = os.environ.get('HDF5_DIR')                                           
HDF5_incdir = os.environ.get('HDF5_INCDIR')                                     
HDF5_libdir = os.environ.get('HDF5_LIBDIR')                                     
netCDF4_dir = os.environ.get('NETCDF4_DIR')                                     
netCDF4_incdir = os.environ.get('NETCDF4_INCDIR')                               
netCDF4_libdir = os.environ.get('NETCDF4_LIBDIR')                               
szip_dir = os.environ.get('SZIP_DIR')                                           
szip_libdir = os.environ.get('SZIP_LIBDIR')                                     
szip_incdir = os.environ.get('SZIP_INCDIR')                                     
hdf4_dir = os.environ.get('HDF4_DIR')                                           
hdf4_libdir = os.environ.get('HDF4_LIBDIR')                                     
hdf4_incdir = os.environ.get('HDF4_INCDIR')                                     
jpeg_dir = os.environ.get('JPEG_DIR')                                           
jpeg_libdir = os.environ.get('JPEG_LIBDIR')                                     
jpeg_incdir = os.environ.get('JPEG_INCDIR')                                     
curl_dir = os.environ.get('CURL_DIR')                                           
curl_libdir = os.environ.get('CURL_LIBDIR')                                     
curl_incdir = os.environ.get('CURL_INCDIR')    
```
Some of these even specify which version is required:
```python
'HDF5 version >= 1.8.0 is required'
```

File: var/spack/repos/builtin/packages/py-netcdf/package.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+class PyNetcdf(PythonPackage):
+    """Python interface to the netCDF Library."""
+    homepage = "http://unidata.github.io/netcdf4-python"
+    url      = "https://github.com/Unidata/netcdf4-python/tarball/v1.2.3.1rel"
+
+    version('1.2.3.1', '4fc4320d4f2a77b894ebf8da1c9895af')
+
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-cython', type=('build', 'run'))
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netcdf/package.py]: Pretty sure this is only needed at build-time, but correct me if I'm wrong.

**Comment by citibeth** [comment]
> I wonder how this package differs from py-netcdf4.

Good catch.

**Pull request closed without merging**